modin/core/io/column_stores/parquet_dispatcher.py

```diff
@@ -215,7 +215,12 @@ def row_groups_per_file(self):
     @property
     def files(self):
         if self._files is None:
-            self._files = self._get_files(self.dataset.files)
+            try:
+                files = self.dataset.files
+            except AttributeError:
+                # compatibility with 4.0.1 <= pyarrow < 8.0.0
+                files = self.dataset._dataset.files
```

Comment:
tested the code locally for the next `pyarrow` versions: 4.0.1, 5.0.0, 6.0.0, 7.0.0, 8.0.0